fuel-p2p/src/orchestrator.rs

```diff
@@ -110,26 +109,157 @@ impl NetworkOrchestrator {
                             P2pRequestEvent::RequestBlock { height, response } => {
                                 let request_msg = RequestMessage::RequestBlock(height);
                                 let channel_item = ResponseChannelItem::ResponseBlock(response);
-                                let _ = self.p2p_service.send_request_msg(None, request_msg, channel_item);
+                                let _ = p2p_service.send_request_msg(None, request_msg, channel_item);
                             },
                             P2pRequestEvent::BroadcastNewBlock { block } => {
                                 let broadcast = GossipsubBroadcastRequest::NewBlock(block);
-                                let _ = self.p2p_service.publish_message(broadcast);
+                                let _ = p2p_service.publish_message(broadcast);
                             },
                             P2pRequestEvent::BroadcastNewTransaction { transaction } => {
                                 let broadcast = GossipsubBroadcastRequest::NewTx(transaction);
-                                let _ = self.p2p_service.publish_message(broadcast);
+                                let _ = p2p_service.publish_message(broadcast);
                             },
                             P2pRequestEvent::BroadcastConsensusVote { vote } => {
                                 let broadcast = GossipsubBroadcastRequest::ConsensusVote(vote);
-                                let _ = self.p2p_service.publish_message(broadcast);
-                            }
+                                let _ = p2p_service.publish_message(broadcast);
+                            },
+                            P2pRequestEvent::Stop => break,
                         }
                     } else {
                         warn!(target: "fuel-libp2p", "Failed to receive P2PRequestEvent");
                     }
                 }
             }
         }
+
+        Ok(self)
+    }
+}
+
+pub struct Service {
+    join: Mutex<Option<JoinHandle<Result<NetworkOrchestrator, anyhow::Error>>>>,
+    network_orchestrator: Arc<Mutex<Option<NetworkOrchestrator>>>,
+    tx_request_event: Sender<P2pRequestEvent>,
+}
+
+impl Service {
+    pub fn new(
```

Comment:
TxPool leverages a different patten for initialization, using a `ServiceBuilder` with an empty new that requires the calling of multiple setter functions before initializing the Service. Moving to something similar here could make it easier to pickup from across the codebase. https://github.com/FuelLabs/fuel-core/blob/master/fuel-txpool/src/service.rs#L24 an example of ServiceBuilder can be seen there

Reply:
presently, I don't see much benefit of using a builder pattern just in order to initialize a service with it

Reply:
The benefit of builder comes from not requiring parameters on intialization, I can see why you want to avoid it but we should probably decide at a higher level if we want the other modules to have it too for consistency

Reply:
Sure, if it makes sense of not having a parameter available at initialization time or some similar case, but in this case there is no need for that. If it makes sense for another module to do it, that should not require the rest to the same if there is no need.